modules/programs/pubs.nix

```
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.pubs;
+
+  configFile = pkgs.writeText ".pubsrc" "${cfg.config}";
+
+in {
+  options.programs.pubs = {
+    enable = mkEnableOption "pubs";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.pubs;
+      defaultText = literalExpression "pkgs.pubs";
+      description = "The package to use for the pubs script.";
+    };
+
+    config = mkOption {
+      type = types.lines;
+      default = "";
+      example = ''
+        [main]
+        pubsdir = ${config.home.homeDirectory}/.pubs
+        docsdir = ${config.home.homeDirectory}/.pubs/doc
+        doc_add = link
+        open_cmd = xdg-open
+
+        [plugins]
+        active = git,alias
+
+        [[alias]]
+
+        [[[la]]]
+        command = list -a
+        description = lists papers in lexicographic order
+
+        [[git]]
+        quiet = True
+        manual = False
+        force_color = False
+      '';
+      description = ''
+        Configuration using syntax written to
+        <filename>$HOME/.pubsrc</filename>.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    home.file.".pubsrc" = mkIf (cfg.config != "") { source = configFile; };
```

You can simplify this (and make it more general in the process) by assigning to the `text` field:
```suggestion
    home.file.".pubsrc" = mkIf (cfg.config != "") { text = cfg.config; };
```